File: frontend/src/types/exam.ts
export type ExamStatus = "draft" | "open" | "closed";

export interface ExamCreateRequest {
  title: string;
  description?: string;
  duration_minutes: number;
  total_score: number;
  pass_score: number;
  question_ids: number[];
}

export interface ExamResponse {
  id: number;
  title: string;
  description?: string;
  duration_minutes: number;
  total_score: number;
  pass_score: number;
  status: ExamStatus;
  question_count: number;
  created_at?: string;
}

// 考试结果相关类型
export interface QuestionResult {
  id: number;
  type: string;
  content: string;
  score: number;
  sort_order: number;
  options?: { id: number; label: string; content: string }[];
  user_answer?: string | null;
  correct_answer: string;
  earned_score?: number | null;
  is_correct?: boolean | null;
  explanation?: string | null;
}

export interface TypeStat {
  type: string;
  total: number;
  correct: number;
  pending: number;
}

export interface AttemptResult {
  id: number;
  exam_id: number;
  exam_title: string;
  status: string;
  started_at: string;
  submitted_at?: string | null;
  duration_minutes: number;
  total_score?: number | null;
  objective_score?: number | null;
  subjective_score?: number | null;
  pass_score: number;
  total_questions: number;
  correct_count: number;
  pending_count: number;
  is_passed?: boolean | null;
  questions: QuestionResult[];
  type_stats: TypeStat[];
}

File: backend/app/schemas/exam.py
"""考试请求/响应模型"""

from datetime import datetime

from pydantic import BaseModel, Field


class ExamCreate(BaseModel):
    """创建考试请求"""
    title: str = Field(..., min_length=1, max_length=200, description="考试标题")
    description: str = Field("", description="考试描述")
    duration_minutes: int = Field(..., ge=1, le=480, description="考试时长（分钟）")
    total_score: float = Field(..., ge=0, description="满分")
    pass_score: float = Field(..., ge=0, description="及格分")
    question_ids: list[int] = Field(..., min_length=1, description="题目 ID 列表")


class ExamUpdate(BaseModel):
    """更新考试请求（所有字段可选）"""
    title: str | None = Field(None, min_length=1, max_length=200)
    description: str | None = None
    duration_minutes: int | None = Field(None, ge=1, le=480)
    total_score: float | None = Field(None, ge=0)
    pass_score: float | None = Field(None, ge=0)


class ExamListItem(BaseModel):
    """考试列表项"""
    id: int
    title: str
    description: str | None = None
    duration_minutes: int
    total_score: float
    pass_score: float
    status: str
    question_count: int
    created_at: datetime

    model_config = {"from_attributes": True}


class ExamQuestionItem(BaseModel):
    """考试中的题目"""
    id: int
    type: str
    content: str
    score: float

    model_config = {"from_attributes": True}
